doc/processes/deprecation.md

```diff
@@ -0,0 +1,63 @@
+<!--
+Copyright (c) 2022, 2022 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] https://openjdk.org/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Eclipse OpenJ9 Deprecation Process
+
+To deprecate a documented OpenJ9 option, behaviour, or feature, follow
+the process described below:
+
+1. Determine the **Java version** the deprecation is to take effect.
+2. Announce the deprecation that is to occur in version n in the
+release notes of version n-1. Furthermore, update any documentation
+to warn about the upcoming changes.
+3. Change the codebase in the head stream following the split for
+version n-1, as well as update any relevant documentation,
+to reflect the deprecation.
+4. Remove any relevant documentation once all versions prior to the
+version n go out of service.
+
+## Important Notes
+
+* Deprecation can only occur when a new Java version is released; this
+should not be confused with a new OpenJ9 version, which may not
+necessarily include a new Java version.
+* Since an OpenJ9 release can include multiple Java versions, the
+deprecation announcement should **only** go in the release notes of the
+Java version that precedes the version in which the deprecation takes
+effect.
+* When the codebase is updated to reflect the deprecation, it will need
+to be `#ifdef`'d so that previous Java versions continue to have the old
+behaviour. As such, any relevant documentation that needs to be updated
+to reflect the deprecation needs to do so by removing the text that
+warns about the upcoming changes, replacing it with a message stating
+that the behaviour is deprecated in version n, and including new text
+describing the new behaviour from version n onward.
+* If the deprecation involves a flip in behaviour that is controlled
```

Comment:
This is true but does it need to be part of the process?

Reply:
No, I don't think it needs to. I had arranged this doc to have the actual policy in the `Eclipse OpenJ9 Deprecation Process` section, and notes or clarifications in the `Important Notes` section. However, if the idea is to view the doc in its entirety as the policy, then I can remove this bullet as well.

Reply:
I think we're better to keep the policy document short and simple.  If we need several notes of clarification, then maybe we need to re-examine the policy.